### ISSUE TITLE
Error message while clicking in the Customize option

### ISSUE THREAD

**Issue by andrelop** [writefreely/writefreely#1110]
### Describe the bug

I'm seeing the following error in my logs when clicking in the `Customize` option  :

```
ERROR: 2024/10/18 20:59:07 collections.go:273: [PROGRAMMER ERROR] WARNING: Collection.hostName is empty! Federation and many other things will fail! If you're seeing this in the wild, please report this bug and let us know what you were doing just before this: https://github.com/writefreely/writefreely/issues/new?template=bug_report.md
```

### Steps to reproduce (if necessary)
1. Click in the option named `Customize`
2. Check the WriteFreely logs
3. See the error message informed above

### Expected behavior
There should be no error messages

### Application configuration
 - **Single mode** 
 - **Database?** mysql
 - **Open registration?** no
 - **Federation enabled?** yes

**Version or last commit**: 0.15.1

## Additional info

Also, when I try to follow the ActivityPub account of my WriteFreely blog, I see in the logs :

```
writefreely-web  | 2024/10/18 21:02:33 Fetching actor https://social.example.com/users/andrelop locally
writefreely-web  | 2024/10/18 21:02:33 Fetching actor https://social.example.com/users/andrelop locally
writefreely-web  | 2024/10/18 21:02:33 Not found; fetching actor https://social.example.com/users/andrelop remotely
writefreely-web  | 2024/10/18 21:02:33 GET https://social.example.com/users/andrelop
writefreely-web  | 2024/10/18 21:02:33 Not found; fetching actor https://social.example.com/users/andrelop remotely
writefreely-web  | 2024/10/18 21:02:33 GET https://social.example.com/users/andrelop
writefreely-web  | 2024/10/18 21:02:34 "GET /api/collections/write.example.com" 200 1.161113ms "gotosocial/0.16.0+git-f1cbf6f (+https://social.example.com)"
writefreely-web  | 2024/10/18 21:02:34 "GET /api/collections/write.example.com" 200 1.161113ms "gotosocial/0.16.0+git-f1cbf6f (+https://social.example.com)"
writefreely-web  | 2024/10/18 21:02:34 "GET /.well-known/webfinger?resource=acct%!A(MISSING)write.example.com%!w(MISSING)rite.example.com" 200 152.407µs "gotosocial/0.16.0+git-f1cbf6f (+https://social.example.com)"
writefreely-web  | 2024/10/18 21:02:34 "GET /.well-known/webfinger?resource=acct%!A(MISSING)write.example.com%!w(MISSING)rite.example.com" 200 152.407µs "gotosocial/0.16.0+git-f1cbf6f (+https://social.example.com)"
writefreely-web  | 2024/10/18 21:02:34 GET 
writefreely-web  | 2024/10/18 21:02:34 GET 
writefreely-web  | ERROR: 2024/10/18 21:02:34 activitypub.go:840: Unable to get actual actor! Get "": unsupported protocol scheme ""
writefreely-web  | ERROR: 2024/10/18 21:02:34 activitypub.go:431: Unable to resolve Follow: Couldn't fetch actual actor.
writefreely-web  | 2024/10/18 21:02:34 "POST /api/collections/andrelop/inbox" 500 547.623813ms "Go-http-client/2.0"
writefreely-web  | ERROR: 2024/10/18 21:02:34 activitypub.go:840: Unable to get actual actor! Get "": unsupported protocol scheme ""
writefreely-web  | ERROR: 2024/10/18 21:02:34 activitypub.go:431: Unable to resolve Follow: Couldn't fetch actual actor.
writefreely-web  | 2024/10/18 21:02:34 "POST /api/collections/andrelop/inbox" 500 547.623813ms "Go-http-client/2.0"
```

My ActivityPub server from which I'm trying to follow the WriteFreely account is based on GoToSocial version 0.16.0.

In the logs above, `social.example.com` is my GoToSocial instance and `write.example.com` is my WriteFreely instance.

**Comment by thebaer** [comment]
It looks like the error message might be a duplicate of #907. Is that the same?

As for the following issue, that might be related to #661, which should've been fixed in v0.15.1 with #1000. Are you still having trouble following from GoToSocial?

**Comment by andrelop** [comment]
> It looks like the error message might be a duplicate of #907. Is that the same?

It seems to be so. Anyway, since it seems to be harmless (only log polution) this is not big deal.

> As for the following issue, that might be related to #661, which should've been fixed in v0.15.1 with #1000. Are you still having trouble following from GoToSocial?

Yes, I was having the same issue. However, I have since updated to GoToSocial 0.17.1 (latest stable release currently) and tried it again.

From the WriteFreely side of things, running with `--debug` I was able to  see the following logs :

```
writefreely-web  | 2024/10/24 12:22:48 Fetching actor https://social.andrelop.org/users/andrelop locally
writefreely-web  | 2024/10/24 12:22:48 Fetching actor https://social.andrelop.org/users/andrelop locally
writefreely-web  | 2024/10/24 12:22:48 Not found; fetching actor https://social.andrelop.org/users/andrelop remotely
writefreely-web  | 2024/10/24 12:22:48 Not found; fetching actor https://social.andrelop.org/users/andrelop remotely
writefreely-web  | 2024/10/24 12:22:48 GET https://social.andrelop.org/users/andrelop
writefreely-web  | 2024/10/24 12:22:48 GET https://social.andrelop.org/users/andrelop
writefreely-web  | 2024/10/24 12:22:48 "GET /api/collections/write.andrelop.org" 200 1.237951ms "gotosocial/0.17.1+git-8a93300 (+https://social.andrelop.org)"
writefreely-web  | 2024/10/24 12:22:48 "GET /api/collections/write.andrelop.org" 200 1.237951ms "gotosocial/0.17.1+git-8a93300 (+https://social.andrelop.org)"
writefreely-web  | 2024/10/24 12:22:48 "GET /.well-known/webfinger?resource=acct%!A(MISSING)write.andrelop.org%!w(MISSING)rite.andrelop.org" 200 147.115µs "gotosocial/0.17.1+git-8a93300 (+https://social.andrelop.org)"
writefreely-web  | 2024/10/24 12:22:48 "GET /.well-known/webfinger?resource=acct%!A(MISSING)write.andrelop.org%!w(MISSING)rite.andrelop.org" 200 147.115µs "gotosocial/0.17.1+git-8a93300 (+https://social.andrelop.org)"
writefreely-web  | 2024/10/24 12:22:48 GET 
writefreely-web  | 2024/10/24 12:22:48 GET 
writefreely-web  | 2024/10/24 12:22:48 "POST /api/collections/andrelop/inbox" 500 385.782314ms "gotosocial/0.17.1+git-8a93300 (+https://social.andrelop.org)"
writefreely-web  | ERROR: 2024/10/24 12:22:48 activitypub.go:840: Unable to get actual actor! Get "": unsupported protocol scheme ""
writefreely-web  | ERROR: 2024/10/24 12:22:48 activitypub.go:431: Unable to resolve Follow: Couldn't fetch actual actor.
writefreely-web  | 2024/10/24 12:22:48 "POST /api/collections/andrelop/inbox" 500 385.782314ms "gotosocial/0.17.1+git-8a93300 (+https://social.andrelop.org)"
writefreely-web  | ERROR: 2024/10/24 12:22:48 activitypub.go:840: Unable to get actual actor! Get "": unsupported protocol scheme ""
```

On the GoToSocial instance logs, I saw the following logs once I try to follow the account on WirteFreely :

```
gotosocial_application  | timestamp="24/10/2024 09:27:06.938" func=server.init.func1.Logger.13.1 level=INFO latency="16.107µs" userAgent="Mozilla/5.0 (X11; Linux x86_64; rv:131.0) Gecko/20100101 Firefox/131.0" method=OPTIONS statusCode=204 path=/api/v1/accounts/01J7HBBRRHWV3PG8SXCY1F4ZDS/follow clientIP=172.19.0.1 requestID=7arqxfmj04000fkwh5b0 msg="No Content: wrote 0B"
gotosocial_application  | timestamp="24/10/2024 09:27:06.987" func=workers.(*Processor).ProcessFromClientAPI level=INFO activityType=Create objectType=Follow fromAccount=andrelop msg="processing from client API"
gotosocial_application  | timestamp="24/10/2024 09:27:06.988" func=server.init.func1.Logger.13.1 level=INFO latency="2.935693ms" userAgent="Mozilla/5.0 (X11; Linux x86_64; rv:131.0) Gecko/20100101 Firefox/131.0" method=POST statusCode=200 path=/api/v1/accounts/01J7HBBRRHWV3PG8SXCY1F4ZDS/follow clientIP=172.19.0.1 requestID=d6rqxfmj04000x1xd3f0 msg="OK: wrote 179B"
gotosocial_application  | timestamp="24/10/2024 09:27:06.989" func=federation.(*federatingActor).Send level=INFO msg="send activity Follow via outbox https://social.andrelop.org/users/andrelop/outbox"
gotosocial_application  | timestamp="24/10/2024 09:27:07.197" func=httpclient.(*Client).DoOnce level=INFO method=GET url=https://write.andrelop.org/api/collections/write.andrelop.org#main-key attempt="1" requestID=yyrqxfmj04000nmp38xg pubKeyID=https://social.andrelop.org/users/andrelop/main-key msg="200 OK"
gotosocial_application  | timestamp="24/10/2024 09:27:07.237" func=httpclient.(*Client).DoOnce level=INFO method=GET url=https://write.andrelop.org/api/collections/write.andrelop.org attempt="1" requestID=yyrqxfmj04000nmp38xg pubKeyID=https://social.andrelop.org/users/andrelop/main-key msg="200 OK"
gotosocial_application  | timestamp="24/10/2024 09:27:07.301" func=httpclient.(*Client).DoOnce level=INFO method=GET url=https://write.andrelop.org/.well-known/webfinger?resource=acct%3Awrite.andrelop.org%40write.andrelop.org attempt="1" requestID=yyrqxfmj04000nmp38xg pubKeyID=https://social.andrelop.org/users/andrelop/main-key msg="200 OK"
gotosocial_application  | timestamp="24/10/2024 09:27:07.340" func=httpclient.(*Client).DoOnce level=INFO method=GET url=https://write.andrelop.org/img/avatars/w.png attempt="1" requestID=yyrqxfmj04000nmp38xg pubKeyID=https://social.andrelop.org/users/andrelop/main-key msg="200 OK"
gotosocial_application  | timestamp="24/10/2024 09:27:07.341" func=dereferencing.(*Dereferencer).fetchAccountAvatar level=WARN requestID=yyrqxfmj04000nmp38xg msg="partially loaded attachment: processMediaSafeley: error loading media https://write.andrelop.org/img/avatars/w.png: store: error executing data function: DereferenceMedia: media body exceeds max size 0B"
gotosocial_application  | timestamp="24/10/2024 09:27:07.344" func=server.init.func1.Logger.13.1 level=ERROR latency="216.800353ms" userAgent="Go (WriteFreely/; +https://write.andrelop.org)" method=GET statusCode=500 path=/users/andrelop clientIP=172.19.0.1 pubKeyID=https://write.andrelop.org/api/collections/write.andrelop.org#main-key errors="Error #01: AuthenticateFederatedRequest: error dereferencing account https://write.andrelop.org/api/collections/write.andrelop.org: enrichAccountSafely: error getting account https://write.andrelop.org/api/collections/write.andrelop.org from database after race: sql: no rows in result set\n" requestID=yyrqxfmj04000nmp38xg msg="Internal Server Error: wrote 54B"
gotosocial_application  | timestamp="24/10/2024 09:27:07.383" func=httpclient.(*Client).DoOnce level=ERROR method=POST url=https://write.andrelop.org/api/collections/andrelop/inbox contentType="application/ld+json; profile=\"https://www.w3.org/ns/activitystreams\"" attempt="1" pubKeyID=https://social.andrelop.org/users/andrelop/main-key msg="http response: 500 Internal Server Error"
```

Looking at the GoToSocial database logs as well at the same time I got the following logs :

```
gotosocial_database  | 2024-10-24 12:29:20.374 UTC [428] ERROR:  duplicate key value violates unique constraint "usernamedomain"
gotosocial_database  | 2024-10-24 12:29:20.374 UTC [428] DETAIL:  Key (username, domain)=(write.andrelop.org, write.andrelop.org) already exists.
gotosocial_database  | 2024-10-24 12:29:20.374 UTC [428] STATEMENT:  INSERT INTO "accounts" ("id", "created_at", "updated_at", "fetched_at", "username", "domain", "avatar_media_attachment_id", "avatar_remote_url", "header_media_attachment_id", "header_remote_url", "display_name", "emojis", "fields", "fields_raw", "note", "note_raw", "memorial", "also_known_as_uris", "moved_to_uri", "move_id", "bot", "locked", "discoverable", "uri", "url", "inbox_uri", "shared_inbox_uri", "outbox_uri", "following_uri", "followers_uri", "featured_collection_uri", "actor_type", "private_key", "public_key", "public_key_uri", "public_key_expires_at", "sensitized_at", "silenced_at", "suspended_at", "suspension_origin") VALUES ('01JAZ81EEXX4V0F0S791HVW433', '2024-10-24 12:29:20.321469+00:00', '2024-10-24 12:29:20.321469+00:00', '2024-10-24 12:29:20.321469+00:00', 'write.andrelop.org', 'write.andrelop.org', '01JAZ81EJ12JVNS98EBXAZRZCN', 'https://write.andrelop.org/img/avatars/w.png', DEFAULT, DEFAULT, 'write.andrelop.org', NULL, 'null', 'null', '', '', FALSE, NULL, DEFAULT, DEFAULT, FALSE, TRUE, FALSE, 'https://write.andrelop.org/api/collections/write.andrelop.org', 'https://write.andrelop.org/', 'https://write.andrelop.org/api/collections/write.andrelop.org/inbox', DEFAULT, 'https://write.andrelop.org/api/collections/write.andrelop.org/outbox', 'https://write.andrelop.org/api/collections/write.andrelop.org/following', 'https://write.andrelop.org/api/collections/write.andrelop.org/followers', DEFAULT, 'Person', DEFAULT, '{"N":<redcated>,"E":<redacted>}', 'https://write.andrelop.org/api/collections/write.andrelop.org#main-key', DEFAULT, DEFAULT, DEFAULT, DEFAULT, DEFAULT) RETURNING "header_media_attachment_id", "header_remote_url", "moved_to_uri", "move_id", "shared_inbox_uri", "featured_collection_uri", "private_key", "public_key_expires_at", "sensitized_at", "silenced_at", "suspended_at", "suspension_origin"
```

Due to this, I went ahead and deleted the record form the `accounts` table on GoToSocial, tried to follow my WriteFreely account again and it worked 👍🏼
